### PR TITLE
Code style enforcement

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
@LorenzMeier this is what I have in mind for travis-ci astyle enforcement. The list of what's excluded is in Tools/check_code_style.sh
I had to lock it down to a specific version of astyle, otherwise we're going to have problems.

Once this is ready to go we can pull the bulk formatting into this PR.
Here's what it will look like broken up by directory to make it somewhat digestible.
https://github.com/PX4/Firmware/compare/PX4:master...dagar:format?expand=0

File: Tools/fix_code_style.sh [hunk 1]
```diff
@@ -1,4 +1,13 @@
-#!/bin/sh
+#!/bin/bash
+
+ASTYLE_VER=`astyle --version`
+ASTYLE_VER_REQUIRED="Artistic Style Version 2.05.1"
```
Comment (LorenzMeier) [inline Tools/fix_code_style.sh]: I think if we want to be so picky (and I agree that we have to control the version) then we would need to build it from source on each platform off a git repo. Else we need some sort of soft matching.

Comment (dagar) [inline Tools/fix_code_style.sh]: https://github.com/dagar/astyle/releases/tag/2.05.1

I can transfer ownership of this repo to PX4 and update the error message to point users at the right version.

Comment (dagar) [inline Tools/fix_code_style.sh]: I've transferred astyle repo to PX4. https://github.com/PX4/astyle

**Comment by dagar** [comment]
@LorenzMeier this is ready to go and only failing because it's catching formatting errors (as intended). 

How do you want to handle getting everything formatted? Massive change to get it over with or smaller changes spread out?

**Comment by LorenzMeier** [comment]
It looks good, I'm just coordinating with people how to phase this in best. I should have a good answer by early next week.

File: Tools/check_code_style.sh [hunk 2]
```diff
@@ -1,16 +1,22 @@
 #!/usr/bin/env bash
 set -eu
 failed=0
-for fn in $(find . -path './src/lib/uavcan' -prune -o \
-                   -path './src/lib/mathlib/CMSIS' -prune -o \
-                   -path './src/lib/eigen' -prune -o \
-                   -path './src/modules/attitude_estimator_ekf/codegen' -prune -o \
-                   -path './NuttX' -prune -o \
+for fn in $(find src/examples \
+                 src/systemcmds \
```
Comment (dagar) [inline Tools/check_code_style.sh]: @LorenzMeier this is where you'd add paths to enforce astyle

**Comment by dagar** [comment]
Updated to only enforce astyle in src/examples and src/systemcmds for now.

**Pull request merged**